### PR TITLE
feat(): add if-validator

### PR DESCRIPTION
**Pull request by develite98**
Hi, `ifValidator` or `ifAsyncValidator` are simple utility functions for help to simplify change dynamically validation of Angular Reactive Form.

### CODE REVIEW

File: docs/src/content/docs/utilities/if-validator.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+`ifValidator` or `ifAsyncValidator` are simple utility functions for help to change dynamically validation of Angular Reactive Form
+
+```ts
+import { ifValidator } from 'ngxtension/if-validation';
+```
+
+## Usage
+
+`ifValidator` accepts a callback condition and validatorFn or validatorFn[].
+
```
Comment (nartc) [inline docs/src/content/docs/utilities/if-validator.md:15]:
```suggestion
`ifValidator` accepts a callback condition and `ValidatorFn` or `ValidatorFn[]`.
```

File: libs/ngxtension/if-validator/project.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	"$schema": "../../../node_modules/nx/schemas/project-schema.json",
+	"projectType": "library",
+	"sourceRoot": "libs/ngxtension/if-validator/src",
+	"targets": {
+		"test": {
+			"executor": "@nx/jest:jest",
+			"outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
+			"options": {
+				"jestConfig": "libs/ngxtension/jest.config.ts",
+				"passWithNoTests": true
```
Comment (nartc) [inline libs/ngxtension/if-validator/project.json:12]:
```suggestion
				"jestConfig": "libs/ngxtension/jest.config.ts",
				"testPathPattern": ["if-validator"],
				"passWithNoTests": true
```

File: libs/ngxtension/if-validator/src/if-validator.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		}
+
+		if (validatorFn instanceof Array) {
+			for (let i = 0; i < validatorFn.length; i++) {
+				const result = validatorFn[i](control);
+				if (result) return result;
+			}
+
+			return null;
+		} else {
```
Comment (nartc) [inline libs/ngxtension/if-validator/src/if-validator.ts:28]: nit: unnecessary `else`

**Review by nartc — CHANGES_REQUESTED** [review]
A couple of nits.

Can you add some tests (either unit tests or e2e tests using the `test-app`)?

**Comment by develite98** [comment]
> A couple of nits.
> 
> Can you add some tests (either unit tests or e2e tests using the `test-app`)?

Hello, @nartc I've already fixed comments and added e2e testing, but as I'm not experienced with it (e2e), could you help me check it?

File: apps/test-app/src/app/if-validator/if-validator.component.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { CommonModule } from '@angular/common';
+import { Component } from '@angular/core';
+import { FormControl, ReactiveFormsModule, Validators } from '@angular/forms';
+import { ifValidator } from 'ngxtension/if-validator';
+
+@Component({
+	selector: 'my-app',
```
Comment (nartc) [inline apps/test-app/src/app/if-validator/if-validator.component.ts:7]: nit: you don't need `selector` for routed component.

File: apps/test-app/src/app/if-validator/if-validator.component.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	},
+	template: `
+		<input [formControl]="form" />
+
+		<pre>Is Form Valid: {{ form.valid }}</pre>
+
+		<button (click)="changeCondition()">Change Form Condition</button>
+	`,
+})
+export default class App {
```
Comment (nartc) [inline apps/test-app/src/app/if-validator/if-validator.component.ts:21]:
```suggestion
export default class App {
export default class IfValidator {
```

**Review by nartc — COMMENTED** [review]
Test looks good enough

**Comment by nartc** [comment]
@all-contributors please add @develite98 for code

**Comment by nartc** [comment]
merge #45

**Pull request merged**